web/packages/teleport/src/UnifiedResources/Resources.tsx

```diff
@@ -182,8 +181,18 @@ export function Resources() {
   );
 }
 
-function resourceKey(resource: UnifiedResource) {
-  return `${resource.kind}/${resourceName(resource)}`;
+export function resourceKey(resource: UnifiedResource) {
+  if (resource.kind === 'node') {
+    return `${resource.hostname}/node`;
+  }
+  return `${resource.name}/${resource.kind}`;
+}
+
+export function resourceName(resource: UnifiedResource) {
+  if (resource.kind === 'app' && resource.friendlyName) {
+    return resource.friendlyName;
+  }
+  return resource.kind === 'node' ? resource.hostname : resource.name;
```

Comment:
nit (feel free to ignore): you could use a single style here like

```suggestion
return resource.kind === 'app' && resource.friendlyName
    ? resource.friendlyName
    : resource.kind === 'node'
    ? resource.hostname
    : resource.name;
```

(or use `if`/`else if` style; same for above).

Reply:
i do prefer the if/else style so I switched to that! Thanks for the catch